NitroxClient/MonoBehaviours/Gui/MainMenu/MainMenuMultiplayerPanel.cs

```diff
@@ -76,10 +78,21 @@ private void CreateServerButton(string text, string joinIp, string joinPort, boo
             Destroy(txt.GetComponent<TranslationLiveUpdate>());
             Button multiplayerButtonButton = multiplayerButtonInst.RequireTransform("NewGameButton").GetComponent<Button>();
             multiplayerButtonButton.onClick = new Button.ButtonClickedEvent();
-            multiplayerButtonButton.onClick.AddListener(() =>
+            multiplayerButtonButton.onClick.AddListener(async () =>
             {
+                if (isJoining)
+                {
+                    // Do not attempt to join multiple servers.
+                    return;
+                }
+                isJoining = true;
+                
                 txt.GetComponent<Text>().color = prevTextColor; // Visual fix for black text after click (hover state still active)
-                OpenJoinServerMenu(joinIp, joinPort);
+                await OpenJoinServerMenuAsync(joinIp, joinPort)
+                    .ContinueWith(t =>
+                    {
+                        isJoining = false;
```

Comment:
With false only being set if this completes, if the connection errors out, will isJoining be stuck in true and therefore stop you connecting to another server?

Reply:
No, `ContinueWith` is always called. The `t` variable has the state with which it completed (successful or faulted) but in either case it's not joining anymore.

Reply:
Okay sounds good, I just wanted to check that was how the functionality behaved